highlight_go/main.go

```diff
@@ -63,9 +63,11 @@ func Print(src []byte, w io.Writer, p syntaxhighlight.Printer) error {
 		// TODO: Clean this up.
 		whitespace := string(src[lastOffset:offset])
```

Comment:
As I suspected, and confirmed with local testing, whitespace will often be `""`. I don't think calling p.Print() with a 0-length string makes sense, so I'd rather put it behind an `if whitespace != "" {` guard.

```
...
len(whitespace): 0
len(whitespace): 0
len(whitespace): 0
len(whitespace): 1
len(whitespace): 0
len(whitespace): 0
len(whitespace): 0
len(whitespace): 0
len(whitespace): 3
len(whitespace): 0
len(whitespace): 0
len(whitespace): 1
len(whitespace): 1
len(whitespace): 0
len(whitespace): 0
len(whitespace): 0
len(whitespace): 0
len(whitespace): 0
...
```